source/develop/release-management/features/network/networking-api-security-groups.html.md

```diff
@@ -0,0 +1,352 @@
+| remote_ip_prefix (*) | string | The remote IP prefix that is matched by this security group rule.                                                                                                |
+| description (*)      | string | A human-readable description for the resource. Default is an empty string.                                                                                       |
+
+(*) - optional attributes
+(**) - optional attributes that ***will not*** be implemented.
+
+### OVN ACL table
+
+An OVN ACL has the following attributes:
+
+| name      | type   | description |
+|-----------|--------|-------------|
+| priority  | int    | Range from 0 to 32,767. The smaller the value, the higher the priority                                                                                   |
+| direction | string | Either 'from-lport' or 'to-lport'.                                                                                                                       |
+| match     | string | The match criteria for the ACL. All the L3-L4 information (protocol, ether protocol, ports, ip prefix) is encoded here. Directly relates to OVS matches. |
+| action    | string | Either 'allow', 'allow-related', 'drop', or 'reject'. Currently, 'reject' behaves as 'drop'.                                                             |
+
+### Mapping the rules
+
+Mapping is a two way game: it is needed to translate *from* a networking
+API security group rule *to* an OVN ACL, ***and*** from an OVN ACL to a
+networking API security group rule.
+
+To achieve the latter, the information that is not **directly** encoded
+in the ACL - e.g. protocol, ethertype, ports (both min & max), security
+group id, remote ip prefix, and description - will be duplicated in the
+ACL external IDs.
+
+As the software matures, (most of) this information can eventually be
+extracted from the ACL match clause, but, in the first implementation,
+the plan is to duplicate it.
+
+The security group rule **white list** behavior will be implemented
+through the priority concept; throughout the code two different
+priorities will be used: a lesser priority for **all** ACLs having a
+deny *action* and an higher priority for **all** ACLs having *allow* or
+*allow-related* actions.
+
+## OVN ACL table & openflow pipeline relationship
+
+Take into account that the ACLs entries will translate directly into
+openflow entries in the openflow tables in openvswitch.
+
+### Direction attribute
+
+The ***direction*** attribute in the ACL has a very important impact in
+the openflow pipeline - in which stage of the pipeline processing will
+the rule be evaluated.
+
+As can be seen below, the OpenFlow pipeline consists of two
+different stages: *ingress* and *egress*.
+
+![of-pipeline](../../../../images/features/network/security-groups/openflow-flow.png)
```

Comment:
I ought to somehow mention this image was looted from the openvswitch specification.

Reply:
Please ensure that this does not violate copyright.

Reply:
I'm trying to figure out how can I do that...

How can I be sure that linking to the copyright - Copyright © 2015; Open Networking Foundation - is enough?